### DISCUSSION TITLE
editing asmdude to work with lst files

### DISCUSSION THREAD

**Discussion by AKRichard68** [HJLebbink/asm-dude#154] [General]
Hello,

  I was just wondering if there is an easy way to edit both the original asmdude and asmdude2 to work with lst files.  I was using asmhighlighter but I like asmdude better.  In asmhighlighter all I had to do was edit the package.def file and add 2 lines to it to get it to work with lst files. After editing the package.def file I replaced the original with the edited version, rezipped it then renamed it with the .vsix extension and it worked perfectly, but Ive spent some time going through the files and didnt see where I could add an extension to it.

  Is this possible, and if so, what file would I find the extension list in?

thanks in advance

**Comment by mrpr0phecy** [comment]
Editing Visual Studio Code extensions, like AsmDude or AsmDude2, to support additional file types involves modifying the extension's source code and configuration files. The process may vary depending on the extension's structure and design.

Here's a general guide:

1. **Locate the Language Server Configuration:**
   - Extensions like AsmDude often use Language Server Protocol (LSP) for language support. Check if there's a configuration file related to LSP settings.

2. **Update Configuration for LST Files:**
   - If you find a configuration file, look for the section where file associations or language support is defined. You'll need to add an entry for LST files.

3. **Search for Extension Points:**
   - Examine the extension's source code for places where file types/extensions are registered. This might involve looking at the `package.json` file or other configuration files used by the extension.

4. **Update the File Association:**
   - Add the file association for LST files in the relevant configuration. For example, if there's an array of supported file types, append "lst" to it.

5. **Rebuild or Reload the Extension:**
   - After making changes, you may need to rebuild or reload the extension in Visual Studio Code for the changes to take effect.

It's important to note that modifying extensions may have unintended consequences, and you should be familiar with the extension's code structure and licensing terms before making changes.

Unfortunately, without specific details about the extension's code structure, I can't provide exact instructions. You may need to explore the source code and documentation of AsmDude or AsmDude2 to identify where to make the necessary changes.

If you encounter difficulties or need more specific guidance, consider reaching out to the extension's maintainers or community for support. Additionally, check if there are any user guides or documentation related to extending the functionality of the specific extension you're working with.

**Reply by HJLebbink** [reply]
Thank you for providing such a nicely generated text!
